pipeline_tools/get_analysis_metadata.py

```diff
@@ -18,19 +28,81 @@ def get_workflow_id(analysis_output_path):
     url = analysis_output_path
     calls = url.split('/call-')
     workflow_id = calls[1].split('/')[-1]
+    print('Got analysis workflow UUID: {0}'.format(workflow_id))
     with open('workflow_id.txt', 'w') as f:
         f.write(workflow_id)
     return workflow_id
 
 
+def get_adapter_workflow_id(analysis_output_path):
+    """Parse the adapter workflow id from one of its analysis workflow output paths.
+
+    Args:
+        analysis_output_path (str): Path to workflow output file.
+
+    Returns:
+        workflow_id (str): String giving Cromwell UUID of the adapter workflow.
+    """
+    url = analysis_output_path
+    calls = url.split('/call-')
+    adapter_workflow_id = calls[0].split('/')[-1]
+    print('Got adapter workflow UUID: {0}'.format(adapter_workflow_id))
+    with open('adapter_workflow_id.txt', 'w') as f:
+        f.write(adapter_workflow_id)
+    return adapter_workflow_id
+
+
+def get_adapter_workflow_version(runtime_environment,
+                                 adapter_workflow_id,
+                                 http_requests,
+                                 use_caas=False,
+                                 caas_key_file=None):
+    """Get the version of the adapter workflow from its workflow id and write the version to a file so that it is
+    available outside of the get_analysis task.
+
+    Args:
+        runtime_environment (str): The cromwell environment the workflow was run in.
+        adapter_workflow_id (str): String giving Cromwell UUID of the adapter workflow.
+        http_requests: `http_requests.HttpRequests` instance, a wrapper around requests provides better retry and
+                       logging.
+        use_caas (bool): whether or not to use Cromwell-as-a-Service.
+        caas_key_file (str): path to CaaS service account JSON key file.
+
+    Raises:
+        requests.HTTPError: for 4xx errors or 5xx errors beyond the timeout
+    """
+
+    def log_before(workflowId):
+        print('Getting the version for adapter workflow {}'.format(workflowId))
+
+    if use_caas:
```

Comment:
You can move some of this repeated code that gets the auth, headers and cromwell_url to a separate function or use the ones in cromwell_tools

Reply:
Will look into it, but if it ends up being a lot more work, would you mind me leaving this as another ticket?